### PR TITLE
Add docs on downward api volume plugin

### PR DESCRIPTION
**Pull request by pmorie**
@adellape PTAL

### CODE REVIEW

File: dev_guide/downward_api.adoc [hunk 1]
```diff
@@ -94,3 +94,97 @@ $ oc logs -p dapi-env-test-pod
...
+        items:  <2>
+          - name: "labels" <3>
+            fieldRef:
+              fieldPath: metadata.labels <4>
+----
+
+<1> The `metadata` field of the volume source configures the downward API volume
+<2> The `items` field holds a list of fields to project into the volume
+<3> The name of the file to project the field into
+<2> The selector of the field to project
```
Comment (soltysh) [inline dev_guide/downward_api.adoc]: s/2/4

File: dev_guide/downward_api.adoc [hunk 2]
```diff
@@ -94,3 +94,97 @@ $ oc logs -p dapi-env-test-pod
...
+    rack: rack-123
+  name: dapi-volume-test-pod
+  annotations:
+    annotation1: 345
+    annotation2: 456 
+spec:
+  containers:
+    - name: volume-test-container
+      image: gcr.io/google_containers/busybox
+      command: ["sh", "-c", "cat /etc/labels /etc/annotations"]
```
Comment (soltysh) [inline dev_guide/downward_api.adoc]: Although you're showing that location here, can you put that information about where those files will be places in the initial paragraph or somewhere at the end?

Comment (pmorie) [inline dev_guide/downward_api.adoc]: @soltysh That's actually handled in the 'brief' example section.

**Comment by adellape** [comment]
LGTM

**Pull request merged**